lib/drives/pages/drive_chat_page.dart

```diff
@@ -0,0 +1,158 @@
+    _pendingRides = _drive.pendingRides!.toSet();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    List<Widget> widgets = [];
+    if (_riders.isEmpty && _pendingRides.isEmpty) {
+      widgets.add(const Center(
+        child: Text("No riders or pending rides"),
+      ));
+    } else {
+      if (_riders.isNotEmpty) {
+        List<Widget> riderColumn = [
+          const Text(
+            "Riders",
+            style: TextStyle(
+              fontSize: 24.0,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          const SizedBox(height: 10.0),
+          riderList(),
+        ];
+        widgets.addAll(riderColumn);
+      }
+      widgets.add(const SizedBox(height: 10.0));
+      if (_pendingRides.isNotEmpty) {
+        List<Widget> pendingRidesColumn = [
+          const Text(
+            "Pending Rides",
+            style: TextStyle(
+              fontSize: 24.0,
+              fontWeight: FontWeight.bold,
+            ),
+          ),
+          const SizedBox(height: 10.0),
+        ];
+        pendingRidesColumn.addAll(pendingRidesList());
+        widgets.addAll(pendingRidesColumn);
+      }
+    }
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Drive Chat'),
+      ),
+      body: RefreshIndicator(
+        onRefresh: loadDrive,
+        child: ListView.separated(
+          itemCount: widgets.length,
+          itemBuilder: (context, index) {
+            return widgets[index];
+          },
+          separatorBuilder: (BuildContext context, int index) {
+            return const SizedBox(height: 10);
+          },
+        ),
+      ),
+    );
+  }
+
+  List<Widget> pendingRidesList() {
+    List<Widget> pendingRidesColumn = [];
+    if (_pendingRides.isNotEmpty) {
+      pendingRidesColumn = List.generate(
+        _pendingRides.length,
+        (index) => PendingRideCard(
+          _pendingRides.elementAt(index),
+          reloadPage: loadDrive,
+          drive: _drive,
+        ),
+      );
+    }
+    return pendingRidesColumn;
+  }
+
+  Row profileRow(Profile profile) {
+    return Row(
+      children: [
+        CircleAvatar(
+          child: Text(profile.username[0]),
+        ),
+        const SizedBox(width: 5),
+        Text(profile.username),
+      ],
+    );
+  }
+
+  Widget riderList() {
+    Widget ridersColumn = Container();
+    if (_riders.isNotEmpty) {
+      ridersColumn = Column(
+        children: List.generate(
+          _riders.length,
+          (index) => InkWell(
+            onTap: () => print("Hey"),
+            child: Padding(
+              padding: const EdgeInsets.symmetric(horizontal: 5.0, vertical: 5.0),
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.spaceBetween,
+                children: [
+                  profileRow(_riders.elementAt(index)),
+                  const Icon(
+                    Icons.chat,
+                    color: Colors.black,
```

Comment:
we should try not to use colors, but instead the existing color schemes (important for theming)

Reply:
i'll leave it for now since this gets changed anyways when we do the chat.